waku/factory/waku_callbacks.nim

```diff
@@ -0,0 +1,4 @@
+import ../waku_relay/protocol
+
+type WakuCallbacks* = ref object
```

Comment:
I have the impression that this new type is not needed and might cause confusion with an already existing type: `WakuCallback`

Reply:
Mmm I agree that it can be confused with libwaku's `WakuCallback`, we can change the name to `AppCallbacks` or something like that.

Whether it's needed or not, the reason for its existence is to be able to create a Waku instance and pass different callbacks that we support to set at the application level.

In a following PR I will add one more callback for a different case, and we will probably need to add more in the near future.

So instead of passing a different parameter for every callback we support to set at the application level, I thought it was easier to have a type with all the callbacks that we support and pass only that object, instead of one new parameter for every new app level callback which can get pretty long and messy.

Any idea on a different solution to this?

Reply:
Okay perfect, I'm happy with `AppCallbacks`